diff --git a/lib/portability.h b/lib/portability.h
--- a/lib/portability.h
+++ b/lib/portability.h
@@ -170,7 +170,16 @@
 // Linux headers not listed by POSIX or LSB
-#ifdef __linux__
+#if defined(__linux__) || (defined(__has_include) && __has_include(<sys/mount.h>))
 #include <sys/mount.h>
+#endif
+#if defined(__linux__) || (defined(__has_include) && __has_include(<sys/statfs.h>))
 #include <sys/statfs.h>
+#else
+struct statfs {
+  unsigned long f_bsize;
+  unsigned long f_frsize;
+};
+#endif
+#ifdef __linux__
 #include <sys/swap.h>
 #include <sys/sysinfo.h>
 #endif
diff --git a/toys.h b/toys.h
--- a/toys.h
+++ b/toys.h
@@ -66,5 +66,11 @@
 // Non-posix headers
 #include <sys/ioctl.h>
+#if __has_include(<sys/syscall.h>)
 #include <sys/syscall.h>
+#elif __has_include(<syscall.h>)
+#include <syscall.h>
+#endif
+#if __has_include(<sys/ttydefaults.h>)
 #include <sys/ttydefaults.h>
+#endif
 
