diff --git a/lib/portability.c b/lib/portability.c
index 6acb0ec..afd4d9a 100644
--- a/lib/portability.c
+++ b/lib/portability.c
@@ -95,6 +95,7 @@ struct mtab_list *xgetmountlist(char *path)
 
 #else
 
+#if __has_include(<mntent.h>)
 #include <mntent.h>
 
 // Check if this type matches list.
@@ -166,6 +167,17 @@ struct mtab_list *xgetmountlist(char *path)
 
   return mtlist;
 }
+#else
+
+int mountlist_istype(struct mtab_list *ml, char *typelist)
+{
+  (void)ml;
+  (void)typelist;
+  return 1;
+}
+
+struct mtab_list *xgetmountlist(char *path) { (void)path; return 0; }
+#endif
 
 #endif
 
diff --git a/lib/tty.c b/lib/tty.c
index 7c249fc..ef450f7 100644
--- a/lib/tty.c
+++ b/lib/tty.c
@@ -16,6 +16,14 @@
 
 #include "toys.h"
 
+#ifndef ECHOCTL
+#define ECHOCTL 0
+#endif
+
+#ifndef ECHOKE
+#define ECHOKE 0
+#endif
+
 // Check stdout, stderr, stdin (in that order) and if none open /dev/tty
 int tty_fd(void)
 {
@@ -103,7 +111,9 @@ void xsetspeed(struct termios *tio, int bps)
   int i = bps2cfspeed(bps);
 
   if (!i) error_exit("unknown speed: %d", bps);
-  cfsetspeed(tio, i);
+  if (cfsetispeed(tio, i) || cfsetospeed(tio, i))
+    perror_exit("cfsetspeed");
+
 }
 
 // Reset terminal to known state, saving copy of old state if old != NULL.
