From: Kevin Alavik <kevin@alavik.se>

diff --git a/lib/portability.c b/lib/portability.c
index 6acb0ec..c2998ba 100644
--- a/lib/portability.c
+++ b/lib/portability.c
@@ -218,6 +230,7 @@ int xnotify_wait(struct xnotify *not, char **path)
 
 #else
 
+#if __has_include(<sys/inotify.h>)
 #include <sys/inotify.h>
 
 struct xnotify *xnotify_init(int max)
@@ -261,6 +274,40 @@ int xnotify_wait(struct xnotify *not, char **path)
   }
 }
 
+#else
+
+struct xnotify *xnotify_init(int max)
+{
+  struct xnotify *not = xzalloc(sizeof(struct xnotify));
+
+  not->max = max;
+  not->kq = -1;
+  not->paths = xmalloc(max * sizeof(char *));
+  not->fds = xmalloc(max * sizeof(int));
+
+  return not;
+}
+
+int xnotify_add(struct xnotify *not, int fd, char *path)
+{
+  if (not->max == not->count) error_exit("xnotify_add overflow");
+  not->fds[not->count] = fd;
+  not->paths[not->count++] = path;
+
+  return 0;
+}
+
+int xnotify_wait(struct xnotify *not, char **path)
+{
+  (void)not;
+  (void)path;
+  errno = ENOSYS;
+
+  return -1;
+}
+
+#endif
+
 #endif
 
 #ifdef __APPLE__
diff --git a/lib/utf8.c b/lib/utf8.c
index d37812c..5ad3760 100644
--- a/lib/utf8.c
+++ b/lib/utf8.c
@@ -1,5 +1,7 @@
 #include "toys.h"
 
+int wcwidth(wchar_t);
+
 // Show width many columns, negative means from right edge, out=0 just measure
 // if escout, send it unprintable chars, otherwise pass through raw data.
 // Returns width in columns, moves *str to end of data consumed.
